### ISSUE TITLE
Agency node segfaults if sent an authenticated HTTP request is sent to its port

### ISSUE THREAD

**Issue by voutilad** [arangodb/arangodb#6583]
## My Environment

* __ArangoDB Version__:        3.3.15
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Cluster
* __Deployment Strategy__:   Manual Start
* __Configuration__:               (see agency config file below)
* __Infrastructure__:               GCE
* __Operating System__:        Ubuntu 16.04
* __Used Package__:              Debian or Ubuntu .deb

## Component, Query & Data

__Affected feature__: Agency service

__AQL query (if applicable)__: none

__AQL explain (if applicable)__: none

__Dataset__: none

__Replication Factor & Number of Shards (Cluster only)__: none

## Steps to reproduce

To start, I'm configuring the Agency nodes in the cluster using the following Ansible template so you can see how mine is configured:
```
uid = arangodb
gid = arangodb
pid-file = /var/run/arangdb-agent.pid

[server]
endpoint = tcp://{{ inventory_hostname }}:5001
jwt-secret = {{ JWT }}
authentication = true
threads = 0
storage-engine = rocksdb
statistics = true

[agency]
my-address = tcp://{{ inventory_hostname }}:5001
{% for host in groups['tag_arangodb'] %}
endpoint = tcp://{{ host }}:5001
{% endfor %}
size = 3
activate = true
supervision = false

[database]
directory = /var/lib/arangodb3/agency

[javascript]
startup-directory = /usr/share/arangodb3/js
app-path = /var/lib/arangodb3-apps

[log]
level = info
output = file:///var/log/arangodb3/arangodb-agent.log
output = syslog://daemon/arangodb-agent

[rocksdb]
enable-statistics = false
```

1. Authenticate by POST'ing username/password JSON to `/_open/auth` on coordinator.
2. Extract JWT from response.
3. Send GET to agency (assume an agency is running on arangodb-1:5001) providing JWT in header like:
```bash
$ dave_taskanalytics_com@arangodb-1:~$ curl -v -H "Authorization: bearer ${JWT}" http://arangodb-1:5001/
*   Trying 10.164.0.2...
* Connected to arangodb-1 (10.164.0.2) port 5001 (#0)
> GET / HTTP/1.1
> Host: arangodb-1:5001
> User-Agent: curl/7.47.0
> Accept: */*
> Authorization: bearer [JWT REMOVED FROM EXAMPLE]
> 
* Empty reply from server
* Connection #0 to host arangodb-1 left intact
curl: (52) Empty reply from server

```
4. And if you look at the logfile for the agency, you'll see messages like
```
Sep 23 18:45:22 arangodb-1 systemd[1]: arangodb-agent.service: Main process exited, code=killed, status=11/SEGV
Sep 23 18:45:22 arangodb-1 systemd[1]: arangodb-agent.service: Unit entered failed state.
Sep 23 18:45:22 arangodb-1 systemd[1]: arangodb-agent.service: Failed with result 'signal'.
```

Confusingly, if I load use a web browser and navigate to the agency port, I get a web interface and login. Attempting to log into the system results in the agency throwing the error:
```
Sep 23 18:56:36 arangodb-1 arangod[17988]: {authentication} This server does not support users
```
But it doesn't segfault.

__Problem__: Agency terminates with segmentation fault.

__Expected result__: Valid HTTP response with some code like 404 or 500...and agency stays running.

If I send the HTTP request without the valid bearer token it doesn't crash the agency instance.

Note: I haven't tested this with authentication turned off.

**Comment by jsteemann** [comment]
@voutilad : thanks for reporting this issue. I just reproduced it locally. Now investigating...

**Comment by voutilad** [comment]
Thanks @jsteemann. It appears the same segfault occurs if sending the same request to the DBServer instance as well.

**Comment by jsteemann** [comment]
@voutilad : yes. It works for coordinators only at the moment, at least whenever I tried it it does crash agency and db server nodes, but no coordinator nodes.
Fix is under way for 3.3.17.

**Comment by graetzer** [comment]
https://github.com/arangodb/arangodb/pull/6605

**Comment by jsteemann** [comment]
Merged back into 3.3 branch. Bugfix will become available with 3.3.17.

**Comment by sleto-it** [comment]
@voutilad, thanks again for reporting this

version 3.3.17 has been released and is available in our website. it includes the fix for this issue

i am now closing this ticket. should the problem persists for you, please let us know and we will reopen

thanks again,
